### PR TITLE
boards: openhwgroup: add CV64A6 on GenesysII board

### PR DESCRIPTION
**Pull request by WorldofJARcraft**
Adds support for the CVA6 CPU on a GenesysII FPGA board with Xilinx AXI Ethernet
(https://github.com/cispa/CVA6-Vivado-Project-with-Xilinx-AXI-Ethernet). The SoC currently contains the CVA6 CPU in 64-bit configuration with the SV39 MMU, interrupt controllers (CLINT and PLIC), UART, a SPI for booting from SD, a boot ROM, and I2C controller for on-board audio, a GPIO and the Xilinx AXI Ethernet subsystem.

### CODE REVIEW

**Comment by WorldofJARcraft** [comment]
Depends on https://github.com/zephyrproject-rtos/zephyr/pull/77732 and https://github.com/zephyrproject-rtos/zephyr/pull/73986

File: boards/openhwgroup/cv64a6_genesysII_cispa/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (nordicjm) [inline boards/openhwgroup/cv64a6_genesysII_cispa/CMakeLists.txt:1]: empty file can go

File: boards/openhwgroup/cv64a6_genesysII_cispa/cv64a6_genesysII_defconfig [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+CONFIG_MDIO_XILINX_AXI_ENET=y
+CONFIG_NET_IPV4=y
+CONFIG_NET_TCP=y
+CONFIG_NETWORKING=y
+
+# required for the DMA to work properly...
+CONFIG_SOC_SERIES_CV64A6_PROVIDE_NONSTANDARD_CACHE_OPTIONS=y
+CONFIG_DMA_XILINX_AXI_DMA_DISABLE_CACHE_WHEN_ACCESSING_SG_DESCRIPTORS=y
+
+# debug
```
Comment (nordicjm) [inline boards/openhwgroup/cv64a6_genesysII_cispa/cv64a6_genesysII_defconfig:50]: this file is not acceptable

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Comment by WorldofJARcraft** [comment]
I have cleaned the defconfigs and removed the empty CMakeLists.txt for my three board PRs.

**Comment by Dri99** [comment]
I am currently using it for a school project, if it can help, the fact that the board is called in some places cv64a6_genesysII_cispa and others cv64a6_genesysII without the "_cispa" suffix gave issues in cmake identifying the files. I fixed by renaming files adding the suffix, but consider also to just remove it everywhere

**Comment by WorldofJARcraft** [comment]
> I am currently using it for a school project, if it can help, the fact that the board is called in some places cv64a6_genesysII_cispa and others cv64a6_genesysII without the "_cispa" suffix gave issues in cmake identifying the files. I fixed by renaming files adding the suffix, but consider also to just remove it everywhere

https://github.com/zephyrproject-rtos/zephyr/pull/77792/files and this PR refer to two slightly different configurations.
This one uses the Xilinx Ethernet subsystem, while https://github.com/zephyrproject-rtos/zephyr/pull/77792/files uses the lowRISC Ethernet (which currently does not have a driver in zephyr).
The files are adapted for these two different configurations respectively. You can use the `-b` parameter to `west build` to select which variant you want to build (`-b cv64a6_genesysII` or `-b cv64a6_genesysII_cispa`).

**Pull request closed without merging**